src/scenic/simulators/carla/simulator.py

```diff
@@ -41,22 +44,32 @@ def __init__(self, carla_map, address='127.0.0.1', port=2000, timeout=10,
 		self.record = record  # whether to save images to disk
 
 	def createSimulation(self, scene, verbosity=0):
-		return CarlaSimulation(scene, self.client, self.map, self.timestep,
+		return CarlaSimulation(scene, self.client, self.tm, self.timestep,
 							   render=self.render, record=self.record,
 							   verbosity=verbosity)
 
+	def destroy(self):
+		settings = self.world.get_settings()
+		settings.synchronous_mode = False
+		settings.fixed_delta_seconds = None
+		self.world.apply_settings(settings)
+		self.tm.set_synchronous_mode(False)
+
+		super(CarlaSimulator, self).destroy()
```

Comment:
Any reason not to just use `super().destroy()`?

Reply:
None, updated